Billing.API/Services/Invoice/InvoiceService.cs

```diff
@@ -75,9 +75,34 @@ public async Task<IEnumerable<InvoiceListItem>> GetInvoices(string clientId)
             return items;
         }
 
-        public async Task TestSapConnection()
+        public async Task<string> TestSapConnection()
         {
-            await Task.Run(() => throw new NotImplementedException());
+            try
+            {
+                await using var conn = new HanaConnection(_options.Value.DbConnectionString);
+
+                await conn.OpenAsync();
+
+                var schema = _options.Value.Schema;
+
+                var query = $"select * from {schema}.oeml";
+
+                var da = new HanaDataAdapter(query, conn);
+
+                var dt = new DataTable("Invoices");
+
+                da.Fill(dt);
+
+                await conn.CloseAsync();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex.Message);
```

Comment:
Please, log _the error_, not the message, in the contrary you will loose information.

```suggestion
                _logger.LogError(ex);
```

Reply:
And better yet, the exception and a message as @nreal suggested: 

```suggestion
                _logger.error(ex, "Error testing Hana connection.");
```